src/schematools/types.py

```diff
@@ -1095,6 +1109,16 @@ def from_dict(cls: type[DTS], obj: Json) -> DTS:  # noqa: D102
         )
 
 
+def _name_join(*parts):
+    """Combine items with underscores, and skip empty values."""
+    return "_".join(filter(None, parts))
+
+
+def _name_join_count(*parts):
+    """Tell how many parts are Combine items with underscores, and skip empty values."""
+    return len(list(filter(None, parts)))
```

Comment:
```suggestion
    """Counts the number of underscores that _name_join inserts."""
    return sum(1 for part in parts if part) - 1
```

Reply:
I just copied it. And you'd be surprised:

```
>>> timeit('len(list(filter(None, parts)))', setup='parts=["abc", "def", ""]')
0.3183160940000036
>>> timeit('sum(1 for part in parts if part) - 1', setup='parts=["abc", "def", ""]')
0.40874755899999116
```

Reply:
Those are not equivalent due to the `- 1` in the `sum` version. The number of underscores is one less than the number of parts joined. That's the real point here, the `sum` stuff was just me getting carried away.

Reply:
ahh I see. yes this code is weird in so many ways. the `- 1` should not be there, as the shortened version is combined with another underscore later on. This is really an attempt to cleanup some hard to read code.

Reply:
But then it doesn't match the docstring anymore :)